fineract-core/src/main/java/org/apache/fineract/infrastructure/core/data/DataValidatorBuilder.java

```diff
@@ -1032,7 +1009,7 @@ public DataValidatorBuilder validatePhoneNumber() {
          */
         final String regex = "^\\+?[0-9. ()-]{0,25}$";
         final Pattern pattern = Pattern.compile(regex);
-        final Matcher matcher = pattern.matcher(this.value.toString());
+        final Matcher matcher = pattern.matcher(this.value != null ? this.value.toString() : "");
```

Comment:
this extra condition is useless, the null check was done a couple lines above. Also even if we are remove that check, using empty string might not be the wanted behaviour!